### PR TITLE
Support Python3 for discovery module

### PR DESCRIPTION
**Pull request by Parkayun**
Hi.
I start supporting Python3.
First of all, discovery module is first. And I'll keep supporting another modules.

Thanks.

### CODE REVIEW

File: googleapiclient/discovery.py [hunk 1]
```diff
@@ -757,7 +774,7 @@ def method(self, **kwargs):
     docs.append('Args:\n')
 
   # Skip undocumented params and params common to all methods.
-  skip_parameters = rootDesc.get('parameters', {}).keys()
+  skip_parameters = list(rootDesc.get('parameters', {}).keys())
```
Comment (methane) [inline googleapiclient/discovery.py]: This should be `list(rootDesc.get('parameters', {}))`.

File: googleapiclient/discovery.py [hunk 2]
```diff
@@ -329,7 +343,10 @@ def _media_size_to_long(maxSize):
     The size as an integer value.
   """
   if len(maxSize) < 2:
-    return 0L
+    try:
+      return eval('0L')
+    except SyntaxError:
+      return 0
```
Comment (methane) [inline googleapiclient/discovery.py]: Just `return 0`

File: googleapiclient/discovery.py [hunk 3]
```diff
@@ -253,8 +262,13 @@ def build_from_document(
...
   future = {}
 
-  if isinstance(service, basestring):
-    service = json.loads(service)
+  try:
+    if isinstance(service, basestring):
+      service = json.loads(service)
+  except NameError:
+    if isinstance(service, str):
+      service = json.loads(service)
```
Comment (methane) [inline googleapiclient/discovery.py]: I don't like this try~except clause.
How about adding this code on top of this module?

```python
try:
    basestring
except NameError:
    basestring = str
```

Comment (Parkayun) [inline googleapiclient/discovery.py]: great idea. but I change like this 
```Python
# Support Python3
import sys
if sys.version_info.major > 2:
  basestring = str
```

File: googleapiclient/discovery.py (hunk 1 above)
Comment (Parkayun) [inline googleapiclient/discovery.py]: Thanks.

File: googleapiclient/discovery.py (hunk 2 above)
Comment (Parkayun) [inline googleapiclient/discovery.py]: Any have some reason for using long literal(return 0L)? @craigcitro

**Comment by craigcitro** [comment]
i think i'd prefer to use `six` rather than play games with imports. (this is the recommended approach from what i saw in the py3 porting guide?)

File: googleapiclient/discovery.py (hunk 2 above)
Comment (craigcitro) [inline googleapiclient/discovery.py]: i suspect the goal was having a consistent type on the return value.

**Comment by Parkayun** [comment]
@craigcitro Oh I forgot the six. I'll commit using six.

**Comment by Parkayun** [comment]
@craigcitro I fixed using six.

**Comment by nathanielmanistaatgoogle** [comment]
I think that this pull request may no longer be necessary as of cf939a33f6b8 - please confirm and take action?

**Pull request closed without merging**